c7n/policy.py

```diff
@@ -1122,26 +1122,47 @@ def validate(self):
             a.validate()
 
     def get_variables(self, variables=None):
-        """Get runtime variables for policy interpolation.
+        """
+        Get C7n environment and runtime variables for policy interpolation.
 
-        Runtime variables are merged with the passed in variables
+        These variables are then merged with the passed in variables
         if any.
         """
         # Global policy variable expansion, we have to carry forward on
         # various filter/action local vocabularies. Where possible defer
         # by using a format string.
         #
         # See https://github.com/cloud-custodian/cloud-custodian/issues/2330
-        if not variables:
-            variables = {}
+
+        result = {}
+
+        # First we'll deal with environment variables, since they have the lowest
+        # precedence.
+        env_variable_prefix = 'C7N_VAR_'
```

Comment:
nit: would be nice to extract this to a function get_env_variables()  instead of inlining the extraction.